### ISSUE TITLE
Roll training of head posture abnormal

### ISSUE THREAD

**Issue by zlyxm** [takuya-takeuchi/FaceRecognitionDotNet#133]
## 1. 300w-lp-roll-krls_0.001_0.1.dat  Training file size comparison ：

dataset 4000
Roll ：**240MB**， Pitch：14.9MB， Yaw：23.3MB

dataset 5000
Roll ：**337MB**， Pitch：18.4MB， Yaw：27.3MB

dataset 8000
Roll ：**694MB**， Pitch：28.1MB， Yaw：38,1MB

![image](https://user-images.githubusercontent.com/27178240/91510365-73837e00-e90f-11ea-8e3e-ab871727957d.png)

**Obviously 300w-lp-roll-krls_0.001_0.1.dat is more than ten to twenty times larger than Pitch or Yaw**

## 2. Roll testing num_wrong is too large：
>[Head Pose Training](https://github.com/takuya-takeuchi/FaceRecognitionDotNet/tree/master/tools/HeadPoseTraining)
> testing num_right: 872
> testing num_wrong: 128
>  testing accuracy:  0.8720

and me

> testing num_right: 951
> testing num_wrong: 49
>  testing accuracy:  0.9510
> train accuracy: 0.9998, test accuracy: 0.9510

**But Pitch and Yaw testing accuracy are both greater than 0.96**

## 3. The actual test roll error is too large
![demo-pic1 jpg_rep](https://user-images.githubusercontent.com/27178240/91512319-58673d00-e914-11ea-9add-a5e16f39b550.jpg)

![download jpg_rep](https://user-images.githubusercontent.com/27178240/91512331-5d2bf100-e914-11ea-8422-fa8cf5c372b9.jpg)

![demo-card-5 jpg_rep](https://user-images.githubusercontent.com/27178240/91512340-60bf7800-e914-11ea-8f42-52d38125bebf.jpg)

> Pitch and Yaw are very easy to use, but Roll is wrong

I don't know where the problem is, can you help me?

**Comment by takuya-takeuchi** [comment]
Roll is very hard to detect exactly.
So machine learning Krls could haave a hard time.
This issue happens for me too.

**Comment by zlyxm** [comment]
yes,Maybe the data set is insufficient. Maybe too many key points cause interference. Generally speaking, when we recognize the roll of a face, we only need to judge whether the eyes are parallel, and whether the center point of the eyes and the chin form an inverted triangle, etc.

**Comment by veersamrat** [comment]
I need these data set all .dat files. could you help me ?

**Comment by aalexvi** [comment]
> I need these data set all .dat files. could you help me ?

find you the files? i can´t find it

**Comment by takuya-takeuchi** [comment]
https://github.com/takuya-takeuchi/FaceRecognitionDotNet/tree/master/tools/HeadPoseTraining#2-download-train-and-test-data

**Comment by Hafsa-Rhanami** [comment]
Where can i find those files : 300w-lp-roll-krls_0.001_0.1.dat & 300w-lp-pitch-krls_0.001_0.1.dat & 300w-lp-yaw-krls_0.001_0.1.dat ??

**Comment by Hafsa-Rhanami** [comment]
> https://github.com/takuya-takeuchi/FaceRecognitionDotNet/tree/master/tools/HeadPoseTraining#2-download-train-and-test-data

Pls where can i find those files :  300w-lp-roll-krls_0.001_0.1.dat & 300w-lp-pitch-krls_0.001_0.1.dat & 300w-lp-yaw-krls_0.001_0.1.dat , cuz i didn't find them when i downloaded the zip file here.
![image](https://github.com/takuya-takeuchi/FaceRecognitionDotNet/assets/85495645/e88b8fed-73d3-4ad9-a87b-4710c4c33745)
